homeassistant/helpers/check_config.py

```diff
@@ -181,7 +181,8 @@ def _comp_error(ex: Exception, domain: str, config: ConfigType) -> None:
         if config_schema is not None:
             try:
                 config = config_schema(config)
-                result[domain] = config[domain]
+                if domain in config:
+                    result[domain] = config[domain]
```

Comment:
I think we should show the config is invalid here and not just silent let it pass. This will not effect startop but urge users to cleanup the removed config.

```suggestion
                if domain in config:
                    result[domain] = config[domain]
                else:
                    raise vol.Invalid(f"Config found for {domain} in configration.yaml. "
                                      "YAML support for this domain has been removed. "
                                      "Please remove from your configration.yaml.")
```

test would need to change. E.g.

```python
async def test_removed_yaml_support(hass: HomeAssistant) -> None:
    """Test config validation check with removed CONFIG_SCHEMA without raise if present."""
    mock_integration(
        hass,
        MockModule(domain="bla", config_schema=cv.removed("bla", raise_if_present=False)),
        False
    )
    files = {YAML_CONFIG_FILE: BASE_CONFIG + "bla:\n  platform: demo"}
    with patch("os.path.isfile", return_value=True), patch_yaml_files(files):
        res = await async_check_ha_config_file(hass)
        log_ha_config(res)

        assert res.keys() == {"homeassistant"}
        assert res.error_str.startswith("Invalid config for [bla]: Config found for bla in configration.yaml.")
```

Reply:
This message is still being shown with raise_if_missing=False. I'll send a log tomorrow

Reply:
It will not raise when starting up HA, only when you try to restart, so it should not break startup, But in fact it is an invalid config, that is why it should fail. An alternative be registering an issue in the issue registry.